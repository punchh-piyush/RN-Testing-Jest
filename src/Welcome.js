import React, {useState} from 'react';
import {
  StyleSheet,
  Text,
  View,
  Pressable,
  SafeAreaView,
  TouchableOpacity,
} from 'react-native';

const Welcome = () => {
  const [changedText, setChangedText] = useState(false);

  const changeTextHandler = () => {
    setChangedText(true);
  };

  return (
    <View style={styles.container}>
      <Text style={styles.title}>Test App</Text>
      {!changedText && <Text style={styles.subTitle}>Not Changed!</Text>}
      {changedText && <Text style={styles.subTitle}>Changed!</Text>}
      <TouchableOpacity
        testID="change-btn"
        onPress={changeTextHandler}
        title="Change Text"
        style={styles.btn}
        color="#FFF">
        <Text style={{color: 'black'}}>Change Text</Text>
      </TouchableOpacity>
    </View>
  );
};

export default Welcome;

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    backgroundColor: 'purple',
    padding: 24,
  },
  title: {
    fontSize: 34,
    fontWeight: 'bold',
    color: 'yellow',
  },
  subTitle: {
    fontSize: 20,
    fontWeight: '400',
    marginVertical: 24,
    color: 'white',
  },

  btn: {
    height: 40,
    width: '70%',
    backgroundColor: 'yellow',
    justifyContent: 'center',
    alignItems: 'center',
  },
});
