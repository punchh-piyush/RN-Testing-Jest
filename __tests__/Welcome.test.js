import Welcome from '../src/Welcome';
import {render, fireEvent, screen} from '@testing-library/react-native';

describe('Welcome', () => {
  test('renders "Test App" as a text', () => {
    // Arrange
    render(<Welcome />);

    // Act
    // ... nothing

    // Assert
    const helloWorldElement = screen.getByText('Test App');
    expect(helloWorldElement).toBeTruthy();
  });

  test('renders "Not Changed!" you if the button was NOT clicked', () => {
    render(<Welcome />);

    const outputElement = screen.getByText('Not Changed!');
    expect(outputElement).toBeTruthy();
  });

  test('renders "Changed!" if the button was clicked', () => {
    // Arrange
    render(<Welcome />);

    // Act
    const buttonElement = screen.getByRole('button');
    fireEvent.press(buttonElement);

    // Assert
    const outputElement = screen.getByText('Changed!');
    expect(outputElement).toBeTruthy();
  });

  test('does not render "Not Changed!" if the button was clicked', () => {
    // Arrange
    render(<Welcome />);

    // Act
    const buttonElement = screen.getByRole('button');
    fireEvent.press(buttonElement);

    // Assert
    const outputElement = screen.queryByText('Not Changed!');
    expect(outputElement).toBeNull();
  });
});
