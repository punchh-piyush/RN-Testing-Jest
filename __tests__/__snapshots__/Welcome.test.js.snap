// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#333",
      "borderBottomLeftRadius": 10,
      "borderBottomRightRadius": 10,
      "padding": 24,
    }
  }
>
  <RCTSafeAreaView />
  <Text
    style={
      {
        "color": "yellow",
        "fontSize": 34,
        "fontWeight": "bold",
      }
    }
  >
    Test App
  </Text>
  <Text
    style={
      {
        "color": "white",
        "fontSize": 20,
        "fontWeight": "400",
        "marginVertical": 24,
      }
    }
  >
    Not Changed!
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "yellow",
        "height": 40,
        "justifyContent": "center",
        "opacity": 1,
        "width": "70%",
      }
    }
    testID="change-btn"
  >
    <Text
      style={
        {
          "color": "black",
        }
      }
    >
      Change Text
    </Text>
  </View>
</View>
`;
